pkg/controller/periodic/periodic.go

```diff
@@ -493,6 +504,41 @@ func (c *Controller) createDataGatherAttributeValues() ([]string, insightsv1alph
 	return disabledGatherers, dp
 }
 
+// wasDataGatherSuccessful reads status conditions of the provided "dataGather" "datagather.insights.openshift.io"
+// custom resource and checks whether the data was successfully uploaded or not and updates status accordingly
+func (c *Controller) wasDataGatherSuccessful(dataGather *insightsv1alpha1.DataGather) bool {
+	var dataUploadedCon *metav1.Condition
+	for i := range dataGather.Status.Conditions {
+		con := dataGather.Status.Conditions[i]
+		if con.Type == status.DataUploaded {
+			dataUploadedCon = &con
+		}
+	}
+	statusSummary := controllerstatus.Summary{
+		Operation: controllerstatus.Uploading,
+		Healthy:   true,
+	}
+	if dataUploadedCon == nil {
+		statusSummary.Healthy = false
+		statusSummary.Count = 5
+		statusSummary.Reason = "DataUploadedConditionNotAvailable"
+		statusSummary.Message = fmt.Sprintf("did not find any %q condition in the %s dataGather resource",
+			status.DataUploaded, dataGather.Name)
+		c.statuses["insightsuploader"].UpdateStatus(statusSummary)
+		return false
+	}
+	if dataUploadedCon.Status == metav1.ConditionFalse {
+		statusSummary.Healthy = false
+		statusSummary.Count = 5
+		statusSummary.Reason = dataUploadedCon.Reason
+		statusSummary.Message = dataUploadedCon.Message
+		c.statuses["insightsuploader"].UpdateStatus(statusSummary)
+		return false
+	}
+	c.statuses["insightsuploader"].UpdateStatus(statusSummary)
+	return true
```

Comment:
nit:
I think this could be refactored a bit to be a little clear. Not bad code, only a recommendation.
* You could modify the conditional part from two `if`, to a `if - else if`
* Then not using several returns, but to set `return statusSummary.Healthy` (as it's already set false or true depending on the branch)
* only one `c.statuses["insightsuploader"].UpdateStatus(statusSummary)` before the return

In any case, the code is not that complex as to **need** this refactor.

Reply:
e.g.
```
if dataUploadedCon == nil {
		statusSummary.Healthy = false
		statusSummary.Count = 5
		statusSummary.Reason = "DataUploadedConditionNotAvailable"
		statusSummary.Message = fmt.Sprintf("did not find any %q condition in the %s dataGather resource",
			status.DataUploaded, dataGather.Name)
	} else if dataUploadedCon.Status == metav1.ConditionFalse {
		statusSummary.Healthy = false
		statusSummary.Count = 5
		statusSummary.Reason = dataUploadedCon.Reason
		statusSummary.Message = dataUploadedCon.Message
	}

	c.statuses["insightsuploader"].UpdateStatus(statusSummary)
	return statusSummary.Healthy
```

Reply:
Thanks a lot for the suggestion. I'll take a look.

Reply:
Updated. Please check

Reply:
Thanks, it looks great